hydra-main/src/main/java/com/addthis/hydra/minion/LogUtils.java

```diff
@@ -38,9 +38,10 @@
     /** Streams task log files from newest to oldest. The returned Stream should be closed. */
     public static Stream<Path> streamTaskLogsByName(JobTask task) throws IOException {
         Path logDir = task.logDir.toPath();
-        return Files.list(logDir)
-                    .filter(path -> Files.isRegularFile(path, LinkOption.NOFOLLOW_LINKS))
-                    .sorted(Collections.reverseOrder());
+        try(Stream<Path> stream = Files.list(logDir)) {
+            return stream.filter(path -> Files.isRegularFile(path, LinkOption.NOFOLLOW_LINKS))
+                         .sorted(Collections.reverseOrder());
+        }
```

Comment:
We're returning a Stream here -- it is kind of important that we don't close it prematurely. The javadoc also notes that the caller is responsible for closing it, and most importantly, no (non-fatal) exception can occur that leaves the file stream open. The `IOException` can only occur during the initial `Files.list`, and if it does, then it handles that cleanup internally.